include/service/jobs/sites/site_create_edit_job.js

```diff
@@ -98,7 +93,13 @@ module.exports = function SiteCreateEditJobModule(pb) {
                         site = pb.DocumentCreator.create('site', mySite);
                     }
 
-                    site.hostname = mySite.hostname || site.hostname;
+                    var hostname = mySite.hostname || site.hostname;
+                    // If this site's hostname has been changed, save off a redirectHost for 301 redirects
+                    if ((site && mySite) && (site.hostname !== hostname) && mySite.uid) {
+                        pb.RequestHandler.saveRedirectHost(mySite.hostname, site.hostname, mySite.uid);
```

Comment:
I'm piggy-backing the site save job to update the RequestHandler's redirectHosts object. Is this worth spinning off a separate job?